File: longest-consecutive-sequence.cpp
#include <vector>
#include <unordered_set>

using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans = 0;
        unordered_set<int> hashset(nums.begin(), nums.end());

        for (int num: nums) {
            if (hashset.find(num-1) == hashset.end()) {
                int len = 0;
                while (hashset.find(num) != hashset.end()) {
                    ++num;
                    ++len;
                }
                ans = max(ans, len);
            }
        }

        return ans;
    }
};

File: longest-consecutive-sequence.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans = 0
        hashset = set(nums)

        for num in nums:
            if num - 1 not in hashset:
                length = 0
                while num in hashset:
                    length += 1
                    num += 1
                ans = max(ans, length)

        return ans
